src/lib/holocene/input/input.svelte

```diff
@@ -219,77 +217,126 @@
-  }
-
-  .input-container.invalid {
-    @apply border-red-700 text-red-700;
+    @apply mr-2 flex w-6 cursor-pointer items-center justify-center rounded-full;
   }
 
   .count {
-    @apply invisible mr-2 font-secondary text-sm font-medium text-primary;
+    @apply mx-2 hidden font-secondary text-sm font-medium tracking-widest;
   }
 
-  .count > .warn {
-    @apply text-orange-600;
+  .input:focus ~ .count {
+    @apply block;
   }
 
-  .count > .error {
-    @apply text-red-700;
+  .hint-text {
+    @apply text-xs;
   }
 
-  input:focus + .count {
-    @apply visible;
-  }
+  /* Light theme styles */
+  .input-container.light {
+    @apply border-primary bg-white text-primary focus-within:border-indigo-600 focus-within:shadow-focus focus-within:shadow-indigo-500/50 focus-within:outline-none;
 
-  .hint-text {
-    @apply mt-2 text-xs;
-  }
+    > .input {
+      @apply caret-indigo-600;
+    }
 
-  .hint-text.error,
-  .hint-text.invalid {
-    @apply text-red-700;
-  }
+    &.disabled {
+      @apply border-gray-600 bg-gray-50  text-gray-600;
 
-  /* Light theme styles */
-  .input-container.light,
-  .input-container.light .icon-container,
-  .input-container.light input {
-    @apply bg-white;
-  }
+      > .input {
+        @apply bg-gray-50;
+      }
 
-  .input-container.light .icon-container {
-    @apply text-gray-400;
-  }
+      > .copy-icon-container {
+        @apply border-gray-600 bg-gray-200;
+      }
+    }
 
-  .input-container.light.disabled {
-    @apply border-gray-600 bg-gray-50  text-gray-600;
-  }
+    &.error,
+    &.invalid {
+      @apply border-danger text-danger focus-within:shadow-red-500/50;
 
-  .input-container.light.disabled input {
-    @apply bg-gray-50;
-  }
+      > .input {
+        @apply caret-danger;
+      }
+    }
+
+    > .count > .ok {
+      @apply text-blue-700;
+    }
+
+    > .count > .warn {
+      @apply text-orange-600;
+    }
+
+    > .count > .error {
+      @apply text-danger;
+    }
+
+    > .icon-container {
+      @apply text-gray-400;
+    }
 
-  .input-container.light.disabled .copy-icon-container {
-    @apply border-gray-600 bg-gray-200;
+    + .hint-text.error,
+    + .hint-text.invalid {
+      @apply text-danger;
+    }
   }
 
   /* Dark theme styles */
-  .input-container.dark,
-  .input-container.dark .icon-container,
-  .input-container.dark input,
-  .input-container.dark .copy-icon-container {
-    @apply bg-gray-900 text-white;
+  label.dark,
+  .hint-text.dark {
+    @apply text-white;
   }
 
-  .input-container.dark input {
-    @apply placeholder:text-gray-200;
-  }
+  .input-container.dark {
+    @apply border-gray-400 bg-transparent text-white focus-within:border-indigo-600 focus-within:bg-primary focus-within:shadow-focus focus-within:shadow-indigo-500/50 focus-within:outline-none;
+
+    > .input {
+      @apply caret-indigo-600;
+    }
+
+    &.disabled {
+      @apply border-gray-600 bg-gray-50 text-gray-600;
```

Comment:
Do we want something darker for the background here when `disabled`? Or maybe we can just add opacity instead of a background color?

![Screenshot 2024-01-17 at 9 05 44 AM](https://github.com/temporalio/ui/assets/15069288/032c1629-48ba-48de-a2e9-0bcf4d7c33d6)